posts/telemetry-yasmin.adoc

```diff
@@ -0,0 +1,253 @@
+
+[source, java]
+----
+@WithSpan
+public void add(@SpanAttribute(value = "hostname") String hostname, Properties systemProps) {
+    ...
+}
+----
+
+A span created using the `@WithSpan` annotation can be given a name. For example, the span created link: https://github.com/yasmin-aumeeruddy/mpTelemetry-Demo/blob/main/inventory/src/main/java/io/openliberty/demo/inventory/InventoryManager.java#L58[here] is given the name "list":
+
+[source,java]
+----
+@WithSpan("list")
+public InventoryList list() {
+    ...
+}
+----
+
+Now that spans have been created manually, we can build the inventory service and deploy it to Open Liberty:
+
+`cd inventory`
+
+`mvn liberty:run`
+
+Navigate to the inventory endpoint:
+
+`http://localhost:9081/inventory/systems/localhost`
+
+You should see five spans in the exporter's endpoint: four spans from inventory and one span from system. For example:
+
+image::img/blog/mptelemetry_inventory_manual_span.png[Inventory manual span]
+
+These spans are all part of one single trace which is emitted from the request to the endpoint above. It is therefore easier to identify the source of inaccuracy or latency in a single request by debugging spans individually.  
+
+See link:https://opentelemetry.io/docs/instrumentation/java/manual/[OpenTelemetry's manual instrumentation documentation].
+
+## Java Agent
+
+The OpenTelemetry Java Agent enables Java applications to generate and capture telemetry data automatically using a JAR that can be attached to any Java 8+ application. You can make use of the Java Agent when MicroProfile Telemetry 1.0 is not enabled in your `server.xml`.
+
+You can attach the JAR by adding the following to your `pom.xml`:
+
+[source, xml]
+----
+<!-- Plugin to package opentelemetry java agent -->
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-dependency-plugin</artifactId>
+    <version>3.3.0</version>
+    <executions>
+        <execution>
+            <id>copy</id>
+            <phase>package</phase>
+            <goals>
+                <goal>copy</goal>
+            </goals>
+        </execution>
+    </executions>
+    <configuration>
+        <artifactItems>
+            <artifactItem>
+                <groupId>io.opentelemetry.javaagent</groupId>
+                <artifactId>opentelemetry-javaagent</artifactId>
+                <version>1.19.0</version>
+                <type>jar</type>
+                <outputDirectory>src/main/liberty/config</outputDirectory>
+                <destFileName>opentelemetry-javaagent.jar</destFileName>
+            </artifactItem>
+        </artifactItems>
+    </configuration>
+</plugin>
+----
+
+You can then run Maven with the `package` goal. This will copy the OpenTelemetry Java Agent in to your server config: 
+
+`mvn package` 
+
+add the following line to your `jvm.options` file along with the link:https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/[agent configuration]:
```

Comment:
```suggestion
Add the following line to your `jvm.options` file along with the link:https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/[agent configuration]:
```

This is a little confusing as written- do you also configure the agent in the jvm.options file? Or is config a spearate step/process